src/wp-includes/fonts.php

```diff
@@ -128,44 +128,17 @@ function wp_get_font_dir() {
  * }
  */
 function wp_font_dir( $create_dir = true ) {
-	/*
-	 * Allow extenders to manipulate the font directory consistently.
-	 *
-	 * Ensures the upload_dir filter is fired both when calling this function
-	 * directly and when the upload directory is filtered in the Font Face
-	 * REST API endpoint.
-	 */
-	add_filter( 'upload_dir', '_wp_filter_font_directory' );
-	$font_dir = wp_upload_dir( null, $create_dir, false );
-	remove_filter( 'upload_dir', '_wp_filter_font_directory' );
-	return $font_dir;
-}
-
-/**
- * A callback function for use in the {@see 'upload_dir'} filter.
- *
- * This function is intended for internal use only and should not be used by plugins and themes.
- * Use wp_get_font_dir() instead.
- *
- * @since 6.5.0
- * @access private
- *
- * @param string $font_dir The font directory.
- * @return string The modified font directory.
- */
-function _wp_filter_font_directory( $font_dir ) {
-	if ( doing_filter( 'font_dir' ) ) {
-		// Avoid an infinite loop.
-		return $font_dir;
-	}
+	// This will also create the /uploads directory if it doesn't exist and $create_dir is true.
+	$upload_dir = wp_upload_dir( null, $create_dir, false );
```

Comment:
This doesn't need to create the uploads directory as `wp_mkdir_p()` will do so. It will also create the date path whether it's used or not, eg `2024/05`.

```suggestion
	// Get base uploads directory.
	$upload_dir = wp_upload_dir( null, false, false );
```

Reply:
> This doesn't need to create the uploads directory as wp_mkdir_p() will do so.

Right, `wp_mkdir_p()` will create a missing uploads directory if the fonts directory is under it. This may not be the case as the `font_dir` filter can change the path. 

I agree this shouldn't be a problem. As `wp_font_dir()` is modeled after `wp_upload_dir()` was thinking to try to keep the default arguments intact, but seems it won't make any difference whether is forces `wp_upload_dir()` to not make a directory. Can change that.

> It will also create the date path whether it's used or not, eg 2024/05.

Right. This is how `wp_upload_dir()` works. When the first argument is not set it uses the current time to set the year/month sub-directory. That's the reason the response includes `basedir` and `baseurl`.

Reply:
I'm not sure what you are saying, why does the uploads path need to be created here? 

If `font_dir` is moved out of the uploads directory, it doesn't need to be created. If the default is used, `yyyy/mm` isn't needed and `wp_mkdir_p` will create any directories as required.

Reply:
> If font_dir is moved out of the uploads directory, it doesn't need to be created.

Right. I added that just to keep `wp_upload_dir()` working with its default params. But agree it doesn't seems to make any difference.

Reply:
> Right. I added that just to keep `wp_upload_dir()` working with its default params, similarly to when uploading other files. But agree it doesn't seems to make any difference.

This is why the call to `wp_upload_dir()` shouldn't attempt to create the directory. It will create paths that most likely will not be used.